src/main/java/com/mercuryCyclists/businessIntelligence/service/SaleStreamProcessing.java

```diff
@@ -35,7 +36,7 @@ public Consumer<KStream<?, SaleEvent>> process() {
                 String productName = v.getProductName();
                 String quantity = v.getQuantity().toString();
                 String price = v.getPrice().toString();
-                String newKey = productName + "_" + quantity + "_" + price;
+                String newKey = UUID.randomUUID().toString();
```

Comment:
This is the sketchy part, we dont have anything thats unique

Reply:
Well the key is never used only the value, so I wouldn't stress it

Reply:
needs to be unique, but yea